### PR TITLE
LibJS: Improve parseInt/parseFloat spec compliance

### PR DESCRIPTION
**Pull request by IdanHo**
```
test/built-ins/parseFloat/S15.1.2.3_A2_T10.js ❌ -> ✅
test/built-ins/parseFloat/S15.1.2.3_A2_T3.js  ❌ -> ✅
test/built-ins/parseFloat/S15.1.2.3_A2_T8.js  ❌ -> ✅
test/built-ins/parseFloat/S15.1.2.3_A2_T9.js  ❌ -> ✅
test/built-ins/parseInt/S15.1.2.2_A2_T10.js   ❌ -> ✅
test/built-ins/parseInt/S15.1.2.2_A2_T3.js    ❌ -> ✅
test/built-ins/parseInt/S15.1.2.2_A2_T8.js    ❌ -> ✅
test/built-ins/parseInt/S15.1.2.2_A2_T9.js    ❌ -> ✅
```

### CODE REVIEW

File: Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeNumberFormat.cpp [hunk 1]
```diff
@@ -308,7 +308,7 @@ static ErrorOr<void> parse_number_system_digits(String core_supplemental_path, U
 
 static String parse_identifiers(String pattern, StringView replacement, UnicodeLocaleData& locale_data, NumberFormat& format)
 {
-    static Utf8View whitespace { "\u0020\u00a0\u200f"sv };
+    static Utf8View constexpr whitespace { "\u0020\u00a0\u200f"sv };
```
Comment (alimpfard) [inline Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeNumberFormat.cpp:311]: Please don't start east-constexpr, it's not even allowed there by the standard (see http://eel.is/c++draft/dcl.pre#nt:simple-declaration - `specifier-seq` only allowed before declarator)

File: Userland/Libraries/LibJS/Runtime/GlobalObject.cpp [hunk 2]
```diff
@@ -386,37 +386,63 @@ JS_DEFINE_NATIVE_FUNCTION(GlobalObject::parse_float)
...
+        // a. Set R to 10.
         radix = 10;
     }
 
+    // 10. If stripPrefix is true, then
     if (strip_prefix) {
-        if (s.length() >= 2 && s[0] == '0' && (s[1] == 'x' || s[1] == 'X')) {
-            s = s.substring(2, s.length() - 2);
+        // a. If the length of S is at least 2 and the first two code units of S are either "0x" or "0X", then
+        if (trimmed_view.length() >= 2 && trimmed_view[0] == '0' && (trimmed_view[1] == 'x' || trimmed_view[1] == 'X')) {
```
Comment (alimpfard) [inline Userland/Libraries/LibJS/Runtime/GlobalObject.cpp:431]: how about this
```suggestion
        if (trimmed_view.length() >= 2 && trimmed_view.substring_view(0, 2).equals_ignoring_case("0x"sv)) {
```

or this if you want it to literally say what the spec says?

```suggestion
        if (trimmed_view.length() >= 2 && trimmed_view.substring_view(0, 2).is_any_of("0x"sv, "0X"sv)) {
```

File: Userland/Libraries/LibJS/Runtime/GlobalObject.cpp [hunk 3]
```diff
@@ -374,7 +374,7 @@ JS_DEFINE_NATIVE_FUNCTION(GlobalObject::parse_float)
     if (vm.argument(0).is_number())
         return vm.argument(0);
     auto input_string = TRY(vm.argument(0).to_string(global_object));
-    auto trimmed_string = input_string.trim_whitespace(TrimMode::Left);
+    auto trimmed_string = MUST(trim_string(global_object, js_string(vm, input_string), AK::TrimMode::Left));
```
Comment (alimpfard) [inline Userland/Libraries/LibJS/Runtime/GlobalObject.cpp:377]: the `AK::` qualifier seems to be in excess?

File: Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeNumberFormat.cpp (hunk 1 above)
Comment (IdanHo) [reply Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeNumberFormat.cpp:311]: Hmm, it seems to compile fine, but sure, I'll change it.

File: Userland/Libraries/LibJS/Runtime/GlobalObject.cpp (hunk 3 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/GlobalObject.cpp:377]: yep, my bad.

File: Userland/Libraries/LibJS/Runtime/GlobalObject.cpp (hunk 2 above)
Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/GlobalObject.cpp:431]: sure

File: Userland/Libraries/LibJS/Runtime/GlobalObject.cpp (hunk 3 above)
Comment (linusg) [reply Userland/Libraries/LibJS/Runtime/GlobalObject.cpp:378]: Doesn't appear to be fixed :^)

Comment (IdanHo) [reply Userland/Libraries/LibJS/Runtime/GlobalObject.cpp:378]: oops, I fixed it only in parseInt, fixing

**Pull request merged**